cli/medperf/commands/compatibility_test/validate_params.py

```diff
@@ -0,0 +1,106 @@
+from medperf.exceptions import InvalidArgumentError
+
+
+class CompatibilityTestParamsValidator:
+    """Validates the input parameters to the CompatibilityTestExecution class"""
+
+    def __validate_cubes(self):
+        if not self.model and not self.benchmark_uid:
+            raise InvalidArgumentError(
+                "A model mlcube or a benchmark should at least be specified"
+            )
+
+        if not self.evaluator and not self.benchmark_uid:
+            raise InvalidArgumentError(
+                "A metrics mlcube or a benchmark should at least be specified"
+            )
+
+    def __raise_redundant_data_source(self):
+        msg = "Make sure you pass only one data source: "
+        msg += "either a prepared dataset, a data path and labels path, or a demo dataset url"
+        raise InvalidArgumentError(msg)
+
+    def __validate_prepared_data_source(self):
+        if any(
+            [
+                self.data_path,
+                self.labels_path,
+                self.demo_dataset_url,
+                self.demo_dataset_hash,
+            ]
+        ):
+            self.__raise_redundant_data_source()
+        if self.data_prep:
+            raise InvalidArgumentError(
+                "A data preparation cube is not needed when specifying a prepared dataset"
+            )
+
+    def __validate_data_path_source(self):
+        if not self.labels_path:
+            raise InvalidArgumentError(
+                "Labels path should be specified when providing data path"
+            )
+        if any([self.demo_dataset_url, self.demo_dataset_hash, self.data_uid]):
+            self.__raise_redundant_data_source()
+
+        if not self.data_prep and not self.benchmark_uid:
+            raise InvalidArgumentError(
+                "A data preparation cube should be passed when specifying raw data input"
+            )
+
+    def __validate_demo_data_source(self):
+        if not self.demo_dataset_hash:
+            raise InvalidArgumentError(
+                "The hash of the provided demo dataset should be specified"
+            )
+        if any([self.data_path, self.labels_path, self.data_uid]):
+            self.__raise_redundant_data_source()
+
+        if not self.data_prep and not self.benchmark_uid:
+            raise InvalidArgumentError(
+                "A data preparation cube should be passed when specifying raw data input"
+            )
+
+    def __validate_data_source(self):
```

Comment:
This function appears to be doing two separate things: validating the parameter combination, and assigning the data source. I would separate this into two functions, and inside the compatibility test class, call `self.data_source = self.validator.get_data_source()`